triage/summarize.py

```diff
@@ -193,22 +192,25 @@ def load_failures(builds_file, tests_files):
         { test_name: [{build: gs://foo/bar, name: test_name, failure_text: xxx}, ...], ...}
     """
     builds = {}
-    for build in json.load(open(builds_file)):
-        if not build['started'] or not build['number']:
-            continue
-        for attr in ('started', 'tests_failed', 'number', 'tests_run'):
-            build[attr] = int(build[attr])
-        build['elapsed'] = int(float(build['elapsed']))
-        if 'pr-logs' in build['path']:
-            build['pr'] = build['path'].split('/')[-3]
-        builds[build['path']] = build
+    with open(builds_file) as f:
```

Comment:
Thanks, we should have used this pattern from the beginning.

Reply:
With CPython I just always assumed it was more of a *convention*, but with PyPy buffered writes were actually not being made before subsequent reads causing the tests to fail.